test/e2e/network/dual_stack.go

```diff
@@ -240,11 +242,16 @@ var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
 		validateServiceAndClusterIPFamily(svc, defaultIPFamily)
 
 		// ensure endpoint belong to same ipfamily as service
-		endpoint, err := cs.CoreV1().Endpoints(svc.Namespace).Get(svc.Name, metav1.GetOptions{})
-		if err != nil {
+		if err := wait.PollImmediate(500*time.Millisecond, 10*time.Second, func() (bool, error) {
+			endpoint, err := cs.CoreV1().Endpoints(svc.Namespace).Get(svc.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			validateEndpointsBelongToIPFamily(svc, endpoint, defaultIPFamily)
+			return true, nil
+		}); err != nil {
```

Comment:
@aojea Isn't the error always nil as we never return nil error always inside `wait.PollImmediate`? Maybe we should print the error [here](https://github.com/kubernetes/kubernetes/pull/87201/files#diff-62557e1c28188e811e7529d548cacb8cR248)

Reply:
if timesout it returns an error `ErrWaitTimeout`

Reply:
thanks for confirming!